### PR TITLE
A4A > Feedback: Initial implementation of in-product feedback mechanism

### PR DESCRIPTION
**Pull request by andrii-lysenko**
Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/1246

## Proposed Changes

* This PR does the initial implementation of the in-product feedback mechanism. 

## Why are these changes being made?

* To ask the users for in-product feedback after some events are performed. 

## Testing Instructions

* Open the A4A live link
* Go to the marketplace > Add a referral > Verify the below screen is shown when you are redirected to the Referrals page. Please note the button won't work as expected. This will be implemented in another PR.
* Append the URL with /referrals/dashboard?flags=-a4a-product-feedback > Repeat the steps without refreshing the page and verify the feedback is not shown. 

Screenshot 2024-12-04 at 8 21 35 AM

Screenshot 2024-12-04 at 8 22 05 AM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/components/a4a-feedback/hooks/use-show-a4a-feedback.tsx
Comment (yashwin) [inline client/a8c-for-agencies/components/a4a-feedback/hooks/use-show-a4a-feedback.tsx:1]: We will need to refactor and make some logic changes here. Please ignore it for now

**Comment by jkguidaven** [comment]
When skipping the feedback. The entire page gets reloaded for some reason.

https://github.com/user-attachments/assets/ef01133a-5cf3-4c48-9678-899ff45b78fd

File: client/a8c-for-agencies/components/a4a-feedback/icons/bad.svg
Comment (jkguidaven) [inline client/a8c-for-agencies/components/a4a-feedback/icons/bad.svg:1]: Can we move this to the asset folders so it is more visible for reuse later?

File: client/a8c-for-agencies/components/a4a-feedback/index.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+	return (
+		<div className="a4a-feedback__wrapper">
+			<div className="a4a-feedback__content">
+				<h1 className="a4a-feedback__title">{ title }</h1>
+				<div className="a4a-feedback__description">{ description }</div>
+				<div className="a4a-feedback__questions">
+					<div className="a4a-feedback__question-details">{ questionDetails }</div>
+					<div className="a4a-feedback__experience-selector">
+						<div className="a4a-feedback__experience-label">{ translate( 'Overall' ) }</div>
```
Comment (jkguidaven) [inline client/a8c-for-agencies/components/a4a-feedback/index.tsx]: Nitpicking: This should be `"a4a-feedback__experience-selector-label` if we follow the class name hierarchy.

**Comment by yashwin** [comment]
> When skipping the feedback. The entire page gets reloaded for some reason.

Thanks for the review, @jkguidaven!

This is fixed here, as this PR doesn't handle the logic: https://github.com/Automattic/wp-calypso/pull/97100

**Review by jkguidaven — COMMENTED** [review]
Overall it works great. I have some few comments. Let me know what you think..

File: client/a8c-for-agencies/components/a4a-feedback/icons/bad.svg
Comment (yashwin) [reply client/a8c-for-agencies/components/a4a-feedback/icons/bad.svg:1]: Done: https://github.com/Automattic/wp-calypso/pull/96934/commits/410a348ba47388de8e9aaa1a3db6627c870f9bf3

File: client/a8c-for-agencies/components/a4a-feedback/index.tsx (hunk 1 above)
Comment (yashwin) [reply client/a8c-for-agencies/components/a4a-feedback/index.tsx]: Done: https://github.com/Automattic/wp-calypso/pull/96934/commits/410a348ba47388de8e9aaa1a3db6627c870f9bf3

**Comment by yashwin** [comment]
Thanks for the feedback, @jkguidaven!

I have addressed your comments. Could you please take another look?

**Review by jkguidaven — APPROVED** [review]
Thanks for addressing my comments. LGTM 🚢

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17047427

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @andrii-lysenko for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17047427).

File: client/a8c-for-agencies/components/a4a-feedback/lib/get-a4a-feedback-props.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+					{ args: { email: args?.email } }
+				) as string,
+				questionDetails: translate( 'How was your experience making a referral?' ),
+				ctaText: translate( 'Submit and continue to Dashboard' ),
+			};
+		case 'agency-details-added':
+			return {
+				title: translate( 'Agency details added!' ),
+				description: translate(
+					"Nice job! Your information has been added to the your agency's public profile."
```
Comment (emilyaudela) [inline client/a8c-for-agencies/components/a4a-feedback/lib/get-a4a-feedback-props.ts:30]: Small typo: This should be either "the" or "your," not "**the your**".

Comment (yashwin) [reply client/a8c-for-agencies/components/a4a-feedback/lib/get-a4a-feedback-props.ts:30]: Thanks for reporting. Fixed it here: https://github.com/Automattic/wp-calypso/pull/97397

**Pull request merged**